src/metabase/lib/query.cljc

```diff
@@ -433,3 +434,73 @@
   (-> a-query
       (dissoc a-query :lib/metadata)
       lib.cache/discard-query-cache))
+
+(defn- stage-seq* [dataset-query]
```

Comment:
Don't call a parameter `dataset-query` or `query` if it's not a whole one of those. This is actually receiving fragments like `[:metric {} 1]` references among other things, so call it `x` or maybe `form`. (We do that in a few parts of `lib.convert` for example.)

If you _are_ passing a whole query around, `query` means a pMBQL query in `metabase.lib.*`. Calling the argument `dataset-query` makes me think its (1) legacy MBQL and (2) in need of `normalize`, since it might be super old legacy MBQL.